### ISSUE TITLE
CLI repeatedly tries to check existence of target organization when running `migrate-repo`, even though it returns `200 OK` the first time

### ISSUE THREAD

**Issue by timrogers** [github/gh-gei#944]
## Description

When starting a migration using `gh gei migrate-repo` from a GitHub Enterprise Server instance, we fetch the target organization in GHEC using the REST API in order to check that it exists (`await _targetGithubApi.DoesOrgExist(args.GithubTargetOrg);`).

Even when this request returns a `200 OK` response, it seems to get retried 5 times. After the fifth time, it moves forward and starts the migration using the API.

Here are verbose logs showing this issue:

```
[11:36] [INFO] You are running the latest version of the gei CLI [v0.38]
[11:36] [INFO] Migrating Repo...
[11:36] [INFO] GITHUB SOURCE ORG: restforce
[11:36] [INFO] SOURCE REPO: ghes-test
[11:36] [INFO] GITHUB TARGET ORG: caffeine-software-sandbox
[11:36] [INFO] TARGET REPO:
[11:36] [INFO] WAIT: true
[11:36] [INFO] GHES API URL: https://timrogers-0c5065dc2d6a3aa44.ghe-test.net/api/v3
[11:36] [INFO] Using GitHub Enterprise Server - verifying server version
[11:36] [DEBUG] HTTP GET: https://timrogers-0c5065dc2d6a3aa44.ghe-test.net/api/v3/meta
[11:36] [DEBUG] GITHUB REQUEST ID: 6f5bfe16-0af1-4c7c-8c5c-52763616e755
[11:36] [DEBUG] RESPONSE (OK): {"verifiable_password_authentication":true,"installed_version":"3.8.1"}
[11:36] [INFO] GitHub Enterprise Server version 3.8.1 detected
[11:36] [INFO] Target repo name not provided, defaulting to same as source repo (ghes-test)
[11:36] [DEBUG] HTTP GET: https://api.github.com/repos/caffeine-software-sandbox/ghes-test
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DBBCE:4A94258:642EA0AC
[11:36] [DEBUG] RESPONSE (NotFound): {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/repos#get-a-repository"}
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DBC83:4A94308:642EA0AC
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] Call failed with HTTP OK, retrying...
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DC160:4A947E2:642EA0AD
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] Call failed with HTTP OK, retrying...
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DC932:4A94FC1:642EA0AE
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] Call failed with HTTP OK, retrying...
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DD5B1:4A95C70:642EA0B0
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] Call failed with HTTP OK, retrying...
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DE4C5:4A96BA6:642EA0B4
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] Call failed with HTTP OK, retrying...
[11:36] [DEBUG] HTTP GET: https://api.github.com/orgs/caffeine-software-sandbox
[11:36] [DEBUG] GITHUB REQUEST ID: C85C:4874:49DF762:4A97E5B:642EA0B8
[11:36] [DEBUG] RESPONSE (OK): {"login":"caffeine-software-sandbox","id":113355734,"node_id":"O_kgDOBsGr1g","url":"https://api.github.com/orgs/caffeine-software-sandbox","repos_url":"https://api.github.com/orgs/caffeine-software-sandbox/repos","events_url":"https://api.github.com/orgs/caffeine-software-sandbox/events","hooks_url":"https://api.github.com/orgs/caffeine-software-sandbox/hooks","issues_url":"https://api.github.com/orgs/caffeine-software-sandbox/issues","members_url":"https://api.github.com/orgs/caffeine-software-sandbox/members{/member}","public_members_url":"https://api.github.com/orgs/caffeine-software-sandbox/public_members{/member}","avatar_url":"https://avatars.githubusercontent.com/u/113355734?v=4","description":null,"is_verified":false,"has_organization_projects":true,"has_repository_projects":true,"public_repos":1,"public_gists":0,"followers":0,"following":0,"html_url":"https://github.com/caffeine-software-sandbox","created_at":"2022-09-12T12:24:07Z","updated_at":"2022-09-12T12:24:07Z","type":"Organization","total_private_repos":14,"owned_private_repos":13,"private_gists":0,"disk_usage":33606,"collaborators":1,"billing_email":"tim@caffeinesoftware.com","default_repository_permission":"read","members_can_create_repositories":true,"two_factor_requirement_enabled":false,"members_allowed_repository_creation_type":"all","members_can_create_public_repositories":true,"members_can_create_private_repositories":true,"members_can_create_internal_repositories":false,"members_can_create_pages":true,"members_can_fork_private_repositories":false,"web_commit_signoff_required":false,"members_can_create_public_pages":true,"members_can_create_private_pages":true,"plan":{"name":"free","space":976562499,"private_repos":10000,"filled_seats":1,"seats":0},"advanced_security_enabled_for_new_repositories":false,"dependabot_alerts_enabled_for_new_repositories":false,"dependabot_security_updates_enabled_for_new_repositories":false,"dependency_graph_enabled_for_new_repositories":false,"secret_scanning_enabled_for_new_repositories":false,"secret_scanning_push_protection_enabled_for_new_repositories":false,"secret_scanning_push_protection_custom_link_enabled":false,"secret_scanning_push_protection_custom_link":null}
[11:36] [DEBUG] HTTP POST: https://timrogers-0c5065dc2d6a3aa44.ghe-test.net/api/v3/orgs/restforce/migrations
[11:36] [DEBUG] HTTP BODY: {"repositories":["ghes-test"],"exclude_metadata":true}
[11:36] [DEBUG] GITHUB REQUEST ID: 577dde02-d178-4e3e-bcd4-6ce21b7064c9
```

I've got the logs, but after the final line, we see a response from the "Start a migration" API, and then the migration moves forward,

**Comment by jankoada** [comment]
Is there a way to revert to the previous version in the meantime? Our migration times have more than tripled with this issue, therefore this puts our prod migration on hold.

**Comment by timrogers** [comment]
> Is there a way to revert to the previous version in the meantime? Our migration times have more than tripled with this issue, therefore this puts our prod migration on hold.

@jankoada There is indeed! You can uninstall by running `gh extension remove github/gh-gei` and then install an old version with `gh extension install github/gh-gei --pin v0.37` (I believe v0.37 isn't affected!). That said, we do plan to get this issue fixed ASAP - hopefully today.

**Comment by dylan-smith** [comment]
Released in [v0.40](https://github.com/github/gh-gei/releases/tag/v0.40)
